webpack/ForemanTasks/Components/common/ClickConfirmation/ClickConfirmation.test.js

```diff
@@ -1,44 +1,252 @@
+      expect(confirmButton).toBeEnabled();
+    });
+
+    it('disables confirm button when checkbox is unchecked', () => {
+      render(<ClickConfirmation {...defaultProps} />);
+
+      const checkbox = screen.getByRole('checkbox');
+      const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+
+      // Check the checkbox first
+      fireEvent.click(checkbox);
+      expect(confirmButton).toBeEnabled();
+
+      // Uncheck the checkbox
+      fireEvent.click(checkbox);
+      expect(checkbox).not.toBeChecked();
+      expect(confirmButton).toBeDisabled();
+    });
+  });
+
+  describe('Button Actions', () => {
+    it('calls onClick and setModalClosed when confirm button is clicked', () => {
+      const onClick = jest.fn();
+      const setModalClosed = jest.fn();
+
+      render(
+        <ClickConfirmation
+          {...defaultProps}
+          onClick={onClick}
+          setModalClosed={setModalClosed}
+        />
+      );
+
+      const checkbox = screen.getByRole('checkbox');
+      const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+
+      // Enable the button by checking the checkbox
+      fireEvent.click(checkbox);
+
+      // Click the confirm button
+      fireEvent.click(confirmButton);
+
+      expect(onClick).toHaveBeenCalledTimes(1);
+      expect(setModalClosed).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls setModalClosed when cancel button is clicked', () => {
+      const setModalClosed = jest.fn();
+
+      render(
+        <ClickConfirmation {...defaultProps} setModalClosed={setModalClosed} />
+      );
+
+      const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+      fireEvent.click(cancelButton);
+
+      expect(setModalClosed).toHaveBeenCalledTimes(1);
+    });
+
+    it('does not call onClick when confirm button is clicked while disabled', () => {
+      const onClick = jest.fn();
+
+      render(<ClickConfirmation {...defaultProps} onClick={onClick} />);
+
+      const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+
+      // Try to click the disabled button
+      fireEvent.click(confirmButton);
+
+      expect(onClick).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('Modal Close Behavior', () => {
+    it('resets checkbox state when modal is closed via confirm', () => {
+      const onClick = jest.fn();
+      const setModalClosed = jest.fn();
+
+      render(
+        <ClickConfirmation
+          {...defaultProps}
+          onClick={onClick}
+          setModalClosed={setModalClosed}
+        />
+      );
+
+      const checkbox = screen.getByRole('checkbox');
+      const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+
+      // Check the checkbox
+      fireEvent.click(checkbox);
+      expect(checkbox).toBeChecked();
+
+      // Click confirm
+      fireEvent.click(confirmButton);
+
+      expect(onClick).toHaveBeenCalled();
+      expect(setModalClosed).toHaveBeenCalled();
+    });
+  });
+});
```

Comment:
```suggestion
```
Duplicate of "calls onClick and setModalClosed when confirm button is clicked"

Reply:
Thanks, removed one of them